src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs

```diff
@@ -137,15 +137,25 @@ public class VmrUpdater : VmrManagerBase, IVmrUpdater
         }
         else
         {
-            await UpdateRepositoryInternal(
-                dependencyUpdate,
-                reapplyVmrPatches: true,
-                additionalRemotes,
-                readmeTemplatePath,
-                tpnTemplatePath,
-                generateCodeowners,
-                discardPatches,
-                cancellationToken);
+            try
+            {
+                var patchesToReapply = await UpdateRepositoryInternal(
+                        dependencyUpdate,
+                        reapplyVmrPatches: true,
+                        additionalRemotes,
+                        readmeTemplatePath,
+                        tpnTemplatePath,
+                        generateCodeowners,
+                        discardPatches,
+                        cancellationToken);
+                return true;
+            }
+            catch (EmptySyncException e)
+            {
+                // TODO: Clean up the work branch?
```

Comment:
There's a TODO here, not sure if you wanted to resolve it in this PR

Reply:
Ah, thanks for noticing. Yeah this TODO doesn't belong here but in a higher level. Removed